solutions/instances/variables.tf

```diff
@@ -0,0 +1,270 @@
+########################################################################################################################
+# Common variables
+########################################################################################################################
+
+variable "ibmcloud_api_key" {
+  type        = string
+  description = "The API Key to use for IBM Cloud."
+  sensitive   = true
+}
+
+variable "existing_resource_group" {
+  type        = bool
+  description = "Whether to use an existing resource group."
+  default     = false
+}
+
+variable "resource_group_name" {
+  type        = string
+  description = "The name of a new or an existing resource group in which to provision resources to."
+}
+
+variable "region" {
+  description = "Region where resources will be created"
```

Comment:
Not necessarily all resources (since you have alias provider blocks for cos and kms). Just observability resources